### PR TITLE
Desktop support poc

### PR DESCRIPTION
**Pull request by joshwooding**
This PR is a POC implementation for #1137 

It tackles runtime style injection as well as window injection. It uses the concept of a component injection too similar to how window in labs works.

Window and styles are currently new packages to ensure we don't introduce any circular dependencies

### CODE REVIEW

File: packages/styles/src/use-style-injection/useStyleInjection.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    return () => {
+      const sheetsMap = windowSheetsMap.get(targetWindow);
+      const style = sheetsMap?.get(id);
+      if (style) {
+        targetWindow.document.head.removeChild(style);
+        sheetsMap?.delete(id);
+      }
+    };
+  });
```
Comment (el-dav) [inline packages/styles/src/use-style-injection/useStyleInjection.ts:50]: Should we have a dependency array? I don't think we need to do this every render

**Comment by shinytoyrobots** [comment]
Is there a reason why we haven't progressed this PR?

**Comment by joshwooding** [comment]
> Is there a reason why we haven't progressed this PR?

This work has been handed to @el-dav and @ImmanuelBaskaran. They are currently investigating the steps forward and setting up dev environments.

**Pull request closed without merging**